### PR TITLE
First l

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Movie entity", e);
+        } finally {
+        session.close();
```
Comment (gigbat) [inline src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: add a tab for readable your code

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaQuery<Movie> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Movie.class);
+            criteriaQuery.from(Movie.class);
+            return session.createQuery(criteriaQuery).getResultList();
+       } catch (Exception e){
+            throw new RuntimeException("Error getting all movies", e);
```
Comment (gigbat) [inline src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: Custom exception should be extended from RuntimeException;

Check the checkstyle file

File: src/main/java/com/dev/theater/model/Movie.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.dev.theater.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
```
Comment (gigbat) [inline src/main/java/com/dev/theater/model/Movie.java:9]:
```suggestion
@Entity
@Table (name = "movie")
```

File: src/main/java/com/dev/theater/model/Movie.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+public class Movie {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    long id;
```
Comment (gigbat) [inline src/main/java/com/dev/theater/model/Movie.java]:
```suggestion
    Long id;
```

File: src/main/java/com/dev/theater/service/impl/MovieServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.dev.theater.library.Service;
+import com.dev.theater.model.Movie;
+import com.dev.theater.service.MovieService;
+
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+    @Inject
+    MovieDao movieDao;
```
Comment (gigbat) [inline src/main/java/com/dev/theater/service/impl/MovieServiceImpl.java]:
```suggestion
    private MovieDao movieDao;
```

**Review by gigbat — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/model/Movie.java (hunk 4 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Movie.java]: done

File: src/main/java/com/dev/theater/model/Movie.java (hunk 3 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Movie.java:9]: done

File: src/main/java/com/dev/theater/service/impl/MovieServiceImpl.java (hunk 5 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/service/impl/MovieServiceImpl.java]: done

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: done

**Review by gigbat — APPROVED** [review]

File: src/main/java/com/dev/theater/dao/MovieDao.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.dev.theater.dao;
+
+import com.dev.theater.library.Dao;
+import com.dev.theater.model.Movie;
+import java.util.List;
+
+@Dao
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/dao/MovieDao.java]: should we use Dao annotation above interface?

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new CrudException("Can't insert Movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new CrudException("Can't insert Movie entity " + movie, e);
```

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        } finally {
+            session.close();
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: let's add check for `null` value

File: src/main/java/com/dev/theater/model/Movie.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+@Entity
+@Table(name = "movie")
+public class Movie {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String description;
+
+    public long getId() {
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/model/Movie.java]:
```suggestion
    public Long getId() {
```

File: src/main/java/com/dev/theater/model/Movie.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        return id;
+    }
+
+    public void setId(long id) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/model/Movie.java]:
```suggestion
    public void setId(Long id) {
```

File: src/main/java/com/dev/theater/util/HibernateUtil.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new CrudException("Error creating Session Factory", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/util/HibernateUtil.java]: it isn't crud problem, let's throw runtime exception here

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/util/HibernateUtil.java (hunk 12 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/util/HibernateUtil.java]: done

File: src/main/java/com/dev/theater/model/Movie.java (hunk 10 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Movie.java]: done

File: src/main/java/com/dev/theater/model/Movie.java (hunk 11 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/model/Movie.java]: done

File: src/main/java/com/dev/theater/dao/MovieDao.java (hunk 6 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/MovieDao.java]: deleted

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java (hunk 7 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: changed

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java (hunk 8 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: changed

File: src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java (hunk 9 above)
Comment (vlykhvar) [reply src/main/java/com/dev/theater/dao/impl/MovieDaoImpl.java]: changed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**